src/amo/components/LandingPage/styles.scss

```diff
@@ -9,6 +9,12 @@
   @include margin-end(20px);
 }
 
+.LandingPage-icon--extension {
+  [dir=rtl] & {
```

Comment:
This could use a comment explaining why we are doing this only on rtl. Something like: `// In LTR, the spacing actually looks correct because of the icon itself has built-in margins in its design.`